source/dofs/dof_tools_constraints.inst.in

```diff
@@ -29,6 +29,20 @@ for (deal_II_dimension : DIMENSIONS; deal_II_space_dimension : SPACE_DIMENSIONS)
       const FullMatrix<double> &,
       const std::vector<unsigned int> &,
       const double);
+    template void DoFTools::make_periodicity_constraints(
+      const DoFHandler<deal_II_dimension,
+                       deal_II_space_dimension>::face_iterator &,
+      const DoFHandler<deal_II_dimension,
+                       deal_II_space_dimension>::face_iterator &,
+      AffineConstraints<float> &,
+      const ComponentMask &,
+      bool,
+      bool,
+      bool,
+      const FullMatrix<double> &,
+      const std::vector<unsigned int> &,
+      const float);
+
```

Comment:
I suggest to do things differently. Can you add the following code to the `for` loop in line 16: `S : REAL_AND_COMPLEX_SCALARS`, and replace the `AffineConstraints<double>` by `AffineConstraints<S>`. This loops through all `Number` types for which `AffineConstraints` is instantiated https://github.com/dealii/dealii/blob/c02086aecc5e6588047865e2bb4ad058bf95e346/source/lac/affine_constraints.inst.in#L17-L20 and thus is safer than adding one part. Then, delete the code within the `DEAL_II_WITH_COMPLEX_VALUES` that comes immediately next, it will be part of `S` from above. I then suggest to do the same with the next `for` loop, replace the `double` arguments by `S` and remove the part explicitly coded for complex numbers. You can look at how things are done for the hanging node constraints further down in this file https://github.com/dealii/dealii/blob/c02086aecc5e6588047865e2bb4ad058bf95e346/source/dofs/dof_tools_constraints.inst.in#L120-L126

Reply:
It complains about `std::complex<float>` and `std::complex<_Tp>` when I do this and I'm not sure how to fix it. 

It's probably best if you take over here tbh. You would just tell me what to change, which probably takes longer than for you to do it. Should I close the pr?

Reply:
That's a good point. Let me open an alternative PR (and thanks for trying, this cut down my effort a bit).